File: config/options.go
package config

import (
	"fmt"
	"os"

	"stylelintbc/cli"
	"stylelintbc/options"
	"stylelintbc/utils"
)

func ConfigureOptions() {
	options := options.GetOptions()

	if len(os.Args) != 2 {
		cli.ShowHelp(options)

		os.Exit(0)
	}

	for _, option := range options {
		if os.Args[1] == option.Name || os.Args[1] == option.Alias {
			option.Handler()

			os.Exit(0)
		}
	}

	utils.WriteMessage(utils.WriteMessageProps{
		Type:    "error",
		Message: "The option you've tried to execute doesn't exist",
	})

	fmt.Println()

	cli.ShowHelp(options)
}

File: src/config/options.rs
use std::env;
use std::process;

use crate::cli::show_help;
use crate::enums::MessageType;
use crate::options::get_options;
use crate::utils::{write_message, WriteMessageProps};

pub fn configure_options() {
    let args: Vec<String> = env::args().collect();
    let options = get_options();

    if args.len() != 2 {
        show_help(options);

        process::exit(0);
    }

    let input = &args[1];

    for option in options {
        if input == option.name || input == option.alias {
            (option.handler)();

            process::exit(0);
        }
    }

    write_message(WriteMessageProps {
        message_type: MessageType::Error,
        message: "The option you've tried to execute doesn't exist",
    });

    println!();

    let options = get_options();

    show_help(options);
}
